hydrator-common/src/main/java/io/cdap/plugin/common/ReferenceBatchSink.java

```diff
@@ -41,4 +46,24 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     IdUtils.validateReferenceName(config.referenceName, collector);
     collector.getOrThrowException();
   }
+
+  /**
+   * Record field-level lineage for batch sink plugins (WriteOperation). This method should be called from prepareRun of
+   * any batch sink plugin.
+   * @param context BatchSinkContext from prepareRun
+   * @param outputName name of output dataset
+   * @param tableSchema schema of fields. Also used to determine list of field names. Schema and schema.getFields() must
+   * not be null.
+   * @param operationName name of the operation
+   * @param description operation description; complete sentences preferred
+   */
+  protected void recordLineage(BatchSinkContext context, String outputName, Schema tableSchema, String operationName,
+                               String description) {
+    LineageRecorder lineageRecorder = new LineageRecorder(context, outputName);
+    lineageRecorder.createExternalDataset(tableSchema);
+    List<String> fieldNames = tableSchema.getFields().stream().map(Schema.Field::getName).collect(Collectors.toList());
```

Comment:
it is better to check if the schema is null, if it is null or the fields is null, if that is true, we should log a message